packages/flutter/lib/src/painting/box_decoration.dart

```diff
@@ -92,6 +92,10 @@ class BoxDecoration extends Decoration {
     this.boxShadow,
     this.gradient,
     this.backgroundBlendMode,
+    @Deprecated(
```

Comment:
We very (like very very) recently decided to try to include some context to help users better understand why we are making a change that requires them to migrate their code. Can you update these deprecation notices to have just a small bit of context, like it being more performant to use ShapeDecoration?

Reply:
Done.

```
@Deprecated(
      'Replace BoxDecoration(shape: BoxShape.circle) with ShapeDecoration(shape: const CircleBorder()). '
      'ShapeDecoration is more efficient than BoxDecoration due to caching. '
      'ShapeDecoration also allows code to be tree-shaken, unlike BoxDecoration.shape. '
      'This feature was deprecated after 3.4.0-34.1.pre.',
    )
```